src/js/bs3/module/HintPopover.js

```diff
@@ -0,0 +1,278 @@
+define([
+  'summernote/base/core/func',
+  'summernote/base/core/list',
+  'summernote/base/core/dom',
+  'summernote/base/core/range'
+], function (func, list, dom, range) {
+  var HintPopover = function (summernote) {
+    var self = this;
+    var ui = $.summernote.ui;
+
+    var $note = summernote.layoutInfo.note;
+    var hint = summernote.options.hint || false;
+
+    if (hint && !(hint instanceof Array)) {
+      hint = [hint];
+    }
+
+    var KEY = {
+      UP: 38,
+      DOWN: 40,
+      ENTER: 13
+    };
+
+    var DROPDOWN_KEYCODES = [KEY.UP, KEY.DOWN, KEY.ENTER];
+
+    var $popover = ui.popover({
+      className: 'note-hint-popover',
+      callback : function ($node) {
+        $node.css({
+          'min-width' : '100px',
+          'padding' : '2px'
+        });
+      }
+    }).render().appendTo('body');
+
+    var $popoverContent = $popover.find('.popover-content');
+
+
+    this.timer = null;
+
+
+    this.events = {
+      'summernote.keyup': function (e, nativeEvent) {
+        self.update(nativeEvent);
+      },
+      'summernote.keydown' : function (e, nativeEvent) {
+        self.updateKeydown(nativeEvent);
+      }
+    };
+
+    this.initialize = function () {
+      if (!hint) {
+        return;
+      }
+      dom.attachEvents($note, this.events);
+
+      $popoverContent.on('click', '.hint-item', function () {
+        $popoverContent.find('.active').removeClass('active');
+        $(this).addClass('active');
+        self.replace();
+        self.hide();
+      });
+    };
+
+    this.destroy = function () {
+      if (!hint) {
+        return;
+      }
+      dom.detachEvents($note, this.events);
+      $popoverContent.off('click');
+    };
+
+    this.activate = function ($activeItem) {
+      $activeItem = $($activeItem);
```

Comment:
$activeItem is not jquery's object.  $activeItem is jquery object or dom element. 

parameter's name is must be changed.